src/basics.md

```diff
@@ -270,6 +271,58 @@ fn run() -> Result<()> {
 # quick_main!(run);
 ```
 
+[ex-regex-filter-log]: #ex-regex-filter-log
+<a name="ex-regex-filter-log"></a>
+## Filter a log file by matching multiple regular expressions
+
+[![regex-badge]][regex] [![cat-text-processing-badge]][cat-text-processing]
+
+Reads a file named `application.log` and only outputs the lines 
+containing “version X.X.X” or some IP address followed by port 443 (e.g. “192.168.0.1:443”).
+
+A [`regex::RegexSet`] is built with [`regex::RegexSetBuilder`].
+Since backslashes are very common in regular expressions, using
+[raw string literals][raw-string-literals] make them more readable.
+
+```rust,no_run
+# #[macro_use]
+# extern crate error_chain;
+extern crate regex;
+
+use std::fs::File;
+use std::io::{BufReader, BufRead};
+
+use regex::RegexSetBuilder;
+
+# error_chain! {
+#     foreign_links {
+#         Io(std::io::Error);
+#         Regex(regex::Error);
+#     }
+# }
+# 
+fn run() -> Result<()> {
+    let log_path = "application.log";
+    let buffered = BufReader::new(File::open(log_path)?);
+
+    let set = RegexSetBuilder::new(&[r#"version "\d\.\d\.\d""#, r#"\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3}:443"#])
```

Comment:
please break the line into shorter ones and run rustfmt on the example.
also please add some third regex for good measure (how about Some warning and some specified text later on?)